src/vcBPA.cpp

```diff
@@ -137,47 +143,40 @@ void vcBPA_Deinit(vcBPAManifold **ppManifold)
   vcBPAManifold *pManifold = *ppManifold;
   *ppManifold = nullptr;
   udWorkerPool_Destroy(&pManifold->pPool);
+  pManifold->nodes.Deinit();
   for (size_t i = 0; i < pManifold->grids.length; ++i)
     pManifold->grids[i].Deinit();
 
   pManifold->grids.Deinit();
   udFree(pManifold);
 }
 
-void vcBPA_AddGrid(vcBPAManifold *pManifold, vdkPointCloud *pModel, udDouble3 center)
+bool vcBPA_NodeHasData(vcBPAManifold *pManifold, vdkPointCloud *pModel, vcBPAOctNode *pNode)
 {
-  if (!pManifold->foundFirstGrid)
-  {
-    vdkPointCloudHeader header = {};
-    if (vdkPointCloud_GetHeader(pModel, &header) != vE_Success)
-      return;
-
-    udDouble4x4 storedMatrix = udDouble4x4::create(header.storedMatrix);
-    udDouble3 boundingBoxCenter = udDouble3::create(header.boundingBoxCenter[0], header.boundingBoxCenter[1], header.boundingBoxCenter[2]);
-    udDouble3 boundingBoxExtents = udDouble3::create(header.boundingBoxExtents[0], header.boundingBoxExtents[1], header.boundingBoxExtents[2]);
-    udDouble3 modelMin = (storedMatrix * udDouble4::create(boundingBoxCenter - boundingBoxExtents, 1.0)).toVector3();
-    udDouble3 modelMax = (storedMatrix * udDouble4::create(boundingBoxCenter + boundingBoxExtents, 1.0)).toVector3();
-
-    if (center.x < modelMin.x || center.y < modelMin.y || center.z < modelMin.z || center.x > modelMax.x || center.y > modelMax.y || center.z > modelMax.z)
-      return;
-  }
+  bool hasData = false;
+  vdkPointBufferF64 *pBuffer;
```

Comment:
Not initialised

Reply:
I'll fix this up, I had planned for it to be a member in `vcBPAManifold` to avoid the cost of creating and destroying it half a million times, but I managed to avoid calling this function for the leaves of the octree instead. The optimisation could still be made, but having vdkQuery support a "ContainsPoints" instead of having to query for a single point would be better.

Reply:
This has been fixed.